packages/clerk-js/src/core/clerk.ts

```diff
@@ -1025,6 +1019,15 @@ export default class Clerk implements ClerkInterface {
         this.#environment = environment;
         this.updateClient(client);
 
+        if (this.isSatellite && this.#options.shouldSyncLink) {
+          if (!this.#hasSynced()) {
```

Comment:
☁️ If make move the code below the next line (updated 983 line)
```
if (this.isSatellite && this.#hasSynced()) {
  this.#clearSynced();
}
```
the `this.#hasSynced()` which is used will always be false even though these lines of code should run as in sync flow.
I don't think we should move it here. Let's discuss it and see where it should be.